include/mpi.h

```diff
@@ -360,28 +371,25 @@ enum {
     MPI_COMBINER_HINDEXED           = 307,
     MPI_COMBINER_INDEXED_BLOCK      = 308,
     MPI_COMBINER_HINDEXED_BLOCK     = 309,
-    MPI_COMBINER_STRUCT             = 310,
-    MPI_COMBINER_SUBARRAY           = 311,
-    MPI_COMBINER_DARRAY             = 312,
-    MPI_COMBINER_F90_REAL           = 313,
-    MPI_COMBINER_F90_COMPLEX        = 314,
-    MPI_COMBINER_F90_INTEGER        = 315,
-    MPI_COMBINER_RESIZED            = 316,
-
-    // File Operation Constants (?)
-    MPI_DISTRIBUTE_NONE         = 404,
-    MPI_DISTRIBUTE_BLOCK        = 401,
-    MPI_DISTRIBUTE_CYCLIC       = 402,
-    MPI_DISTRIBUTE_DFLT_DARG    = 403,
-
+    MPI_COMBINER_SUBARRAY           = 310,
+    MPI_COMBINER_DARRAY             = 311,
+    MPI_COMBINER_VALUE_INDEX        = 312,
+    MPI_COMBINER_STRUCT             = 313,
+    MPI_COMBINER_F90_REAL           = 314,
+    MPI_COMBINER_F90_COMPLEX        = 315,
+    MPI_COMBINER_F90_INTEGER        = 316,
+    MPI_COMBINER_RESIZED            = 317,
```

Comment:
Again, I am curious about the reasons behind the constants shuffling.

Reply:
Lisanda wants MPI_DISTRIBUTE... next to MPI_ORDER... constants.

Reply:
In this case, I went from "simpler" to "more complex" combiners. VALUE_INDEX is simpler, special case of STRUCT, so I listed it first.

Reply:
i think the order i used was how it appears in the spec but we can of course reorder the spec table to match some design principle rather than history

Reply:
> i think the order i used was how it appears in the spec

That's a compelling argument. Moreover, `MPI_Type_get_value_index` in the reductions section of the MPI std, well after all other datatype constructors have been defined.

I'll update the order accordingly. This is basically the same as before, with a new entry for VALUE_INDEX at the end.